processes/consumer/flush.go

```diff
@@ -56,64 +57,76 @@ func Flush(ctx context.Context, inMemDB *models.DatabaseData, dest destination.B
 		go func(_tableName string, _tableData *models.TableData) {
 			defer wg.Done()
 
-			logFields := []any{
-				slog.String("tableName", _tableName),
+			if args.CoolDown != nil && _tableData.ShouldSkipFlush(*args.CoolDown) {
+				slog.Debug("Skipping flush because we are currently in a flush cooldown", slog.String("tableName", _tableName))
+				return
 			}
 
-			if args.CoolDown != nil && _tableData.ShouldSkipFlush(*args.CoolDown) {
-				slog.With(logFields...).Debug("Skipping flush because we are currently in a flush cooldown")
+			action := "merge"
+			if _tableData.Mode() == config.History {
+				action = "append"
+			}
+
+			retryCfg, err := retry.NewJitterRetryConfig(500, 30_000, 10, retry.AlwaysRetry)
+			if err != nil {
+				slog.Error("Failed to create retry config", slog.Any("err", err))
 				return
 			}
 
-			// Lock the tables when executing merge / append.
 			_tableData.Lock()
 			defer _tableData.Unlock()
 			if _tableData.Empty() {
 				return
 			}
 
-			// This is added so that we have a new temporary table suffix for each merge / append.
-			_tableData.ResetTempTableSuffix()
-
 			start := time.Now()
 			tags := map[string]string{
-				"what":     "success",
 				"mode":     _tableData.Mode().String(),
 				"table":    _tableName,
 				"database": _tableData.TopicConfig().Database,
 				"schema":   _tableData.TopicConfig().Schema,
 				"reason":   args.Reason,
 			}
 
-			var err error
-			action := "merge"
-			// Merge or Append depending on the mode.
-			if _tableData.Mode() == config.History {
-				err = dest.Append(_tableData.TableData, false)
-				action = "append"
-			} else {
-				err = dest.Merge(_tableData.TableData)
-			}
+			what, err := retry.WithRetriesAndResult(retryCfg, func(_ int, _ error) (string, error) {
+				return flush(ctx, dest, _tableName, _tableData)
+			})
 
 			if err != nil {
-				tags["what"] = "merge_fail"
-				tags["retryable"] = fmt.Sprint(dest.IsRetryableError(err))
-				slog.With(logFields...).Error(fmt.Sprintf("Failed to execute %s, not going to flush memory, will sleep for 3 seconds before continuing...", action), slog.Any("err", err))
-				time.Sleep(3 * time.Second)
+				slog.Error(fmt.Sprintf("Failed to %s", action), slog.Any("err", err), slog.String("tableName", _tableName))
 			} else {
-				slog.Info(fmt.Sprintf("%s success, clearing memory...", stringutil.CapitalizeFirstLetter(action)), logFields...)
-				commitErr := commitOffset(ctx, _tableData.TopicConfig().Topic, _tableData.PartitionsToLastMessage)
-				if commitErr == nil {
-					inMemDB.ClearTableConfig(_tableName)
-				} else {
-					tags["what"] = "commit_fail"
-					slog.Warn("Commit error...", slog.Any("err", commitErr))
-				}
+				inMemDB.ClearTableConfig(_tableName)
+				slog.Info(fmt.Sprintf("%s success, clearing memory...", stringutil.CapitalizeFirstLetter(action)), slog.String("tableName", _tableName))
```

Comment:
Might want to swap the order of these two lines, or change the message, as "clearing memory..." sort of implies that the memory hasn't been cleared yet.